src/screens/IndexScreen.js

```diff
@@ -0,0 +1,75 @@
+import React, { useContext } from "react";
+import {
+  View,
+  Text,
+  StyleSheet,
+  FlatList,
+  Button,
+  TouchableOpacity,
+} from "react-native";
+import { Entypo } from "@expo/vector-icons";
+import { Feather } from "@expo/vector-icons";
+import { Context } from "../context/BlogContext";
+import { useNavigation } from "@react-navigation/native";
+
+const IndexScreen = () => {
+  const navigation = useNavigation();
+  const { state, addBlogPost, deleteBlogPost } = useContext(Context);
+
+  React.useLayoutEffect(() => {
+    navigation.setOptions({
```

Comment:
This is not a bad solution however I am not a big fan of scattering the navigation into multiple components. Because when your app grows large, it is not one navigation component that you can go to and see what is wrong, you will have navigation spread around in multiple components.

Remember you can always have nested navigators, and those nested navigators can have internal logic on what to show and what not to show. If that showing is based on some state you can always pass that state to the nested navigators.

Read React Navigation documentation, it will help you a lot: you can always have a [main navigator](https://reactnavigation.org/docs/native-stack-navigator), like the one you already have in your `App` component. and then you can have nested navigators like [MaterialBottonTabs](https://reactnavigation.org/docs/material-bottom-tab-navigator) and [MaterialTopTabs](https://reactnavigation.org/docs/material-top-tab-navigator)